components/pages/discovery/components/VisibleElements.tsx

```diff
@@ -0,0 +1,41 @@
+/*
+ *
+ * Copyright (c) 2025 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ *  This program and the accompanying materials are made available under the terms of
+ *  the GNU Affero General Public License v3.0. You should have received a copy of the
+ *  GNU Affero General Public License along with this program.
+ *   If not, see <http://www.gnu.org/licenses/>.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ *  EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ *  OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ *  SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ *  TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ *  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ *  IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+import { useChartsContext } from '@overture-stack/arranger-charts';
+
+/**
+ * Text to display the visible amount of elements for a chart out of total data
+ *
+ * @param maxElements - Max number of chart data elements to show (covers bars and segments)
+ * @param fieldName - Field name of data
+ * @returns Text showing "x of y"
+ */
+export const VisibleElements = ({ maxElements, fieldName }) => {
+  const { getChartData } = useChartsContext();
+  const { isLoading, isError, data } = getChartData(fieldName);
+
+  const noData = isLoading || isError || (data && data?.length === 0);
+  const totalWithinMax = !noData && data.length <= maxElements;
+
+  return noData || totalWithinMax ? null : (
+    <div>{`Top ${Math.min(maxElements, data.length)} of ${data.length}`}</div>
+  );
```

Comment:
This probably works, but is a tough read - I think because it uses negative logic. Maybe it simplifies with:


```suggestion
  const hasData = !isLoading && !isError && data && data.length > 0;
  const hasMoreThanMaxElements = hasData && data.length >= maxElements;

  return hasMoreThanMaxElements ?  (
    <div>{`Top ${Math.min(maxElements, data.length)} of ${data.length}`}</div>
  ) : null;
```